contracts/settings/AlloSettings.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+contract AlloSettings is OwnableUpgradeable {
+
+  string public constant VERSION = "1.0.0";
+
+  uint8 public constant PERCENTAGE_PRECISION = 100;
+
+  // --- Data ---
+
+  /// @notice Address of the protocol treasury
+  address payable public protocolTreasury;
+
+  /// @notice Protocol fee percentage
+  /// 100% = 10000 | 10% = 1000 | 1% = 100 | 0.1% = 10 | 0.01% = 1
+  uint16 public protocolFeePercentage;
+
+  // --- Event ---
+
+  /// @notice Emitted when protocol fee percentage is updated
+  event ProtocolFeePercentageUpdated(uint16 protocolFeePercentage);
+
+  /// @notice Emitted when a protocol wallet address is updated
+  event ProtocolTreasuryUpdated(address protocolTreasuryAddress);
+
+  /// @notice constructor function which ensure deployer is set as owner
+  function initialize() external initializer {
+    __Context_init_unchained();
+    __Ownable_init_unchained();
+  }
+
+  // --- Core methods ---
+
+  /// @notice Set the protocol fee percentage
+  /// @param _protocolFeePercentage The new protocol fee percentage
+  function updateProtocolFeePercentage(uint16 _protocolFeePercentage) external onlyOwner {
```

Comment:
Nit: not sure if it's worth adding a check here? We might not want to cap `_protocolFeePercentage` to an arbitrary number, but still it can be set to > 100%.

Reply:
good call out @DanieleSalatti ! Adding this change + tests